cmd/auth/login.go

```diff
@@ -20,6 +36,23 @@ var loginCmd = &cobra.Command{
 		defer perisistentAuth.Close()
 		ctx, cancel := context.WithTimeout(cmd.Context(), loginTimeout)
 		defer cancel()
+
+		profileFlag := cmd.Flag("profile")
+		if profileFlag != nil && profileFlag.Value.String() != "" {
+			perisistentAuth.Profile = profileFlag.Value.String()
+		} else {
+			profiles, err := getProfilesMap(cmd.Context())
+			if err != nil {
+				return err
+			}
+			if len(profiles) > 0 {
+				profile, err := cmdio.Select(ctx, profiles, "~/.databrickscfg profile")
+				if err != nil {
+					return err
+				}
+				perisistentAuth.Profile = profile
+			}
+		}
 		return perisistentAuth.Challenge(ctx)
```

Comment:
Can you move the profile-saving code out of `Challenge` and into this function? Then it'll be easier to 1) hook up the prompt once we're done, 2) allow use of the configured profile to select the cluster.